### DISCUSSION TITLE
Vite server restart by changing env files do not update env in globalThis.

### DISCUSSION THREAD

**Discussion by hyoban** [wakujs/waku#562] [Ideas]
https://github.com/hyoban-repro/waku-env-repro

1. start dev server
2. change content in `.env` file

**Comment by dai-shi** [comment]
Thanks for reporting. Yeah, that's expected.
Does bare Vite+React handle it well?

**Comment by hyoban** [comment]
> Does bare Vite+React handle it well?

Yes, with `import.meta.env`

**Comment by dai-shi** [comment]
Does Vite hot reload or hard reload if you edit&save `.env` file?

**Comment by dai-shi** [comment]
I'm not sure how we could do it, and if it's a priority for us, but @Aslemammad might have some idea.

**Comment by hyoban** [comment]
> Does Vite hot reload or hard reload if you edit&save `.env` file?

hard reload

**Comment by dai-shi** [comment]
Thanks. I don't think it's a priority, and there are many tasks now, but it's good to know.

**Comment by hyoban** [comment]
Yes, restarting the server is not a troublesome thing for me.

**Comment by Aslemammad** [comment]
I'll look into it, would be happy to be assigned.

**Comment by dai-shi** [comment]
but, less priority please.

**Comment by dai-shi** [comment]
Moving to discussion / ideas for now.

**Comment by dai-shi** [comment]
@Aslemammad fixed it: #798
